### PR TITLE
APPS-1068 [AGS] Publish AGS-specific docker images from acs-packaging

### PR DESCRIPTION
**Pull request by elizastan**

### CODE REVIEW

File: docker-alfresco/ags/pom.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    <packaging>pom</packaging>
+
+    <parent>
+        <groupId>org.alfresco</groupId>
+        <artifactId>content-services-docker</artifactId>
+        <version>7.1.0-SNAPSHOT</version>
+    </parent>
+    <properties>
+        <image.name>alfresco/alfresco-governance-repository-enterprise</image.name>
+        <s3connector.version>3.1.0</s3connector.version>
```
Comment (CezarLeahu) [inline docker-alfresco/ags/pom.xml:16]: Remove S3-related properties from this pom.

File: docker-alfresco/ags/pom.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                            <goal>copy</goal>
+                        </goals>
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>org.alfresco</groupId>
+                                    <artifactId>alfresco-share-services</artifactId>
+                                    <version>${dependency.alfresco-community-repo.version}</version>
+                                    <type>amp</type>
+                                </artifactItem>
```
Comment (CezarLeahu) [inline docker-alfresco/ags/pom.xml:38]: The *Share Services* amp should be removed from this pom as it is already included in the base docker image (check [acs-packaging/blob/master/docker-alfresco/pom.xml#L50-L57](https://github.com/Alfresco/acs-packaging/blob/master/docker-alfresco/pom.xml#L50-L57)).

File: docker-alfresco/ags/pom.xml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                                    <version>${dependency.alfresco-enterprise-repo.version}</version>
+                                    <type>war</type>
+                                    <destFileName>gs-api-explorer-${project.version}.war</destFileName>
+                                </artifactItem>
+                                <artifactItem>
+                                    <groupId>org.alfresco</groupId>
+                                    <artifactId>alfresco-governance-services-community-repo</artifactId>
+                                    <version>${dependency.alfresco-community-repo.version}</version>
+                                    <type>amp</type>
+                                </artifactItem>
```
Comment (CezarLeahu) [inline docker-alfresco/ags/pom.xml:51]: Remove the GS community AMP. The Enterprise AMP is enough.

File: docker-alfresco/ags/pom.xml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                                    <version>${dependency.alfresco-enterprise-repo.version}</version>
+                                    <type>amp</type>
+                                </artifactItem>
+                                <artifactItem>
+                                    <groupId>org.alfresco.integrations</groupId>
+                                    <artifactId>alfresco-s3-connector</artifactId>
+                                    <version>${s3connector.version}</version>
+                                    <type>amp</type>
+                                    <overWrite>false</overWrite>
+                                </artifactItem>
```
Comment (CezarLeahu) [inline docker-alfresco/ags/pom.xml:64]: Remove the S3 amp from this image.

File: docker-alfresco/ags/pom.xml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                                    <artifactId>alfresco-share-services</artifactId>
+                                    <version>${dependency.alfresco-community-repo.version}</version>
+                                    <type>amp</type>
+                                </artifactItem>
+                                <artifactItem>
+                                    <groupId>org.alfresco</groupId>
+                                    <artifactId>alfresco-governance-services-enterprise-rest-api-explorer</artifactId>
+                                    <version>${dependency.alfresco-enterprise-repo.version}</version>
+                                    <type>war</type>
+                                    <destFileName>gs-api-explorer-${project.version}.war</destFileName>
```
Comment (CezarLeahu) [inline docker-alfresco/ags/pom.xml:44]: The *gs-api-explorer* version can remain `${dependency.alfresco-enterprise-repo.version}`.

File: docker-alfresco/ags/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+# Set default docker_context.
+ARG resource_path=target
+
+#Use the root user
+USER root
+
+### Copy the AMP from build context to amps
+COPY target/alfresco-governance-services-enterprise-repo-*.amp /usr/local/tomcat/amps/
+COPY target/amps/alfresco-share-services-*.amp /usr/local/tomcat/amps/
```
Comment (CezarLeahu) [inline docker-alfresco/ags/Dockerfile:12]: Remove the share services installation as it's already part of the base image.

File: docker-alfresco/ags/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+# Install amps on alfresco.war
+RUN java -jar /usr/local/tomcat/alfresco-mmt/alfresco-mmt*.jar install \
+              /usr/local/tomcat/amps \
+              /usr/local/tomcat/webapps/alfresco -directory -nobackup
+### Copy gs-api-explorer war into webapps folder
+COPY target/amps/gs-api-explorer-*.war /usr/local/tomcat/webapps/
```
Comment (CezarLeahu) [inline docker-alfresco/ags/Dockerfile:19]: Change the source path to just: `target/gs-api-explorer-*.war`.

File: docker-share/ags/pom.xml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+                           <version>${dependency.alfresco-enterprise-share.version}</version>
+                           <type>amp</type>
+                        </artifactItem>
+                        <artifactItem>
+                           <groupId>org.alfresco.integrations</groupId>
+                           <artifactId>alfresco-s3-connector</artifactId>
+                           <version>${s3connector.version}</version>
+                           <type>amp</type>
+                           <overWrite>false</overWrite>
+                        </artifactItem>
```
Comment (CezarLeahu) [inline docker-share/ags/pom.xml:46]: Remove the S3 amp from this pom.

File: docker-alfresco/ags/pom.xml [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    <description>Alfresco Governance Services Enterprise Repository Extension</description>
+    <packaging>pom</packaging>
+
+    <parent>
+        <groupId>org.alfresco</groupId>
+        <artifactId>content-services-docker</artifactId>
+        <version>7.1.0-SNAPSHOT</version>
+    </parent>
+    <properties>
+        <image.name>alfresco/alfresco-governance-repository-enterprise</image.name>
```
Comment (ehardon) [inline docker-alfresco/ags/pom.xml:15]: maybe we can have this tag in camel case

**Review by ehardon — APPROVED** [review]

**Review by CezarLeahu — APPROVED** [review]

**Pull request merged**